dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java

```diff
@@ -255,19 +258,54 @@ public void exportToORCID(@Parameter(hidden = true, name = "user") @Auth User us
         if (orcidByUserId.isEmpty()) {
             throw new CustomWebApplicationException("ORCID account is not linked to user account", HttpStatus.SC_BAD_REQUEST);
         }
+        if (optionalVersion.isPresent()) {
+            putCode = optionalVersion.get().getVersionMetadata().getOrcidPutCode();
+        } else {
+            putCode = entry.getOrcidPutCode();
+        }
         try {
-            String orcidWorkString = ORCIDHelper.getOrcidWorkString(entry, optionalVersion);
-            HttpResponse response = ORCIDHelper.postWorkString(baseApiURL, user.getOrcid(), orcidWorkString,
-                    orcidByUserId.get(0).getToken());
-            if (response.getStatusLine().getStatusCode() != HttpStatus.SC_CREATED) {
-                throw new CustomWebApplicationException("Could not export to ORCID: " + response.getStatusLine().getReasonPhrase(), response.getStatusLine().getStatusCode());
+            String orcidWorkString = ORCIDHelper.getOrcidWorkString(entry, optionalVersion, putCode);
+            HttpResponse response;
```

Comment:
Ditto here, think both the client and the response should be closed to be safe. 
Think this implements `CloseableHttpResponse`